change-validation/src/main/scala/com/normation/plugins/changevalidation/NotificationService.scala

```diff
@@ -0,0 +1,237 @@
+          override protected def getPasswordAuthentication = new PasswordAuthentication(l.value, p)
+        }
+        Session.getInstance(prop, auth)
+      case (_, None) =>
+        prop.put("mail.smtp.auth", "false");
+        Session.getInstance(prop, null)
+      case (None, _) =>
+        prop.put("mail.smtp.auth", "false");
+        Session.getInstance(prop, null)
+    }
+    try {
+      val message = new MimeMessage(session);
+      message.setFrom(new InternetAddress(conf.email.value))
+      message.setRecipients(
+        Message.RecipientType.TO,
+        mailParameter.to.map(_.value).mkString(",")
+      )
+      val reply: Array[Address] = mailParameter.replyTo.map(e => new InternetAddress(e.value)).toArray
+      message.addRecipients(Message.RecipientType.BCC, mailParameter.bcc.map(_.value).mkString(","))
+      message.addRecipients(Message.RecipientType.CC, mailParameter.cc.map(_.value).mkString(","))
+      message.setReplyTo(reply)
+      message.setSubject(mailParameter.subject);
+
+      message.setText(content);
+      Transport.send(message)
+      mailParameter.succeed
+    } catch {
+      case e: MessagingException =>
+        logger.error(s"Sending mail to notification have failed, cause by : ${e.getMessage}").fail
+    }
+  }
+
+  def getConfig(path: String): Config = {
+    val file = new File(path)
+    val configResource = if(file.exists && file.canRead) {
+      Some(FileSystemResource(file))
+    } else {
+      logger.error("Can not find configuration file specified by JVM property %s: %s ; abort")
+      throw new Exception("Configuration file not found: %s".format(file.getPath))
+    }
+    ConfigFactory.load(ConfigFactory.parseFile(configResource.value.file))
+  }
+
+
+  //TO-DO have to deal with the case where a parameter is not set (logger)
+  def getSMTPConf(path: String): IO[Throwable, SMTPConf] = {
+    for {
+      config     <- ZIO.fromTry(Try(getConfig(path)))
+      hostServer <- ZIO.fromTry(Try(config.getString("hostServer")))
+      port       <- ZIO.fromTry(Try(config.getInt("hostServer")))
+      email      <- ZIO.fromTry(Try(config.getString("hostServer")))
+      login      <- (Try(config.getString("hostServer")) match {
+                      case Success(l) => Some(Username(l))
+                      case _ =>
+                        logger.warn(s"No `login` defined for SMTP conf in ${path}")
+                        None
+                    }).succeed
+      password   <- (Try(config.getString("hostServer")) match {
+                      case Success(p) => Some(p)
+                      case _ =>
+                        logger.warn(s"No `password` is defined for SMTP conf in ${path}")
+                        None
+                    }).succeed
+    } yield {
+      SMTPConf(
+          hostServer
+        , port
+        , Email(email)
+        , login
+        , password
+      )
+    }
+  }
+
+  def getValidationMailParameter(path: String): IO[Throwable, EmailInfo] = {
```

Comment:
You should have one generic function to get emailConfiguration for a step, 

def GetStepMailConf(step : WorkflowNode) : IOResult[EmailConf] = {
IOResult.effect("Error when getting $step email configuration") {
  to = config.getString("$step-to")
  ...
}
}